iphone/cli/lib/process-js-task.js

```diff
@@ -0,0 +1,374 @@
+	 * @param {String|Array} states Single state string or array of states to filter
+	 * @return {Array<String>}
+	 */
+	filterFilesByStatus(changedFiles, states) {
+		states = Array.isArray(states) ? states : [ states ];
+		const filteredFiles = [];
+		changedFiles.forEach((fileState, filePath) => {
+			if (states.includes(fileState)) {
+				filteredFiles.push(filePath);
+			}
+		});
+		return filteredFiles;
+	}
+
+	/**
+	 * Resolves th given full path to a relative path under it will be available in the app.
+	 *
+	 * @param {String} fullPath Full path used to resolve the relative path.
+	 * @param {Object} jsFiles Optional lookup object. Defaults to this.jsFiles.
+	 * @return {String}
+	 */
+	resolveRelativePath(fullPath, jsFiles) {
+		jsFiles = jsFiles || this.jsFiles;
+		return Object.keys(jsFiles).find(relativePath => {
+			return jsFiles[relativePath].src === fullPath;
+		});
+	}
+
+	/**
+	 * Handle deleted files by cleaning up the internal task data
+	 *
+	 * This will:
+	 *
+	 * - Remove file from the list of bootstrap files (if it's a .bootstrap.js file)
+	 * - Delete it's content hash
+	 * - Delete used Ti symbols
+	 *
+	 * @param {String} filePathAndName Full path to the deleted file
+	 * @return {Promise}
+	 */
+	handleDeletedFile(filePathAndName) {
+		let file = this.resolveRelativePath(filePathAndName, this.data.jsFiles);
+		const bootstrapPath = file.substr(0, file.length - 3);  // Remove the ".js" extension.
+		if (bootstrapPath.endsWith('.bootstrap')) {
+			const index = this.jsBootstrapFiles.indexOf(bootstrapPath);
+			if (index !== -1) {
+				this.jsBootstrapFiles.splice(index, 1);
+			}
+		}
+
+		const info = this.data.jsFiles[file];
+		if (info) {
+			delete this.data.contentHashes[info.src];
+			delete this.data.tiSymbols[info.dest];
+		}
+
+		return Promise.resolve();
+	}
+
+	/**
+	 * Performs some sanity checks if we can safely perform an incremental build
+	 * or need to fallback to a full build.
+	 *
+	 * @return {Boolean} True if a full build is required, false if not.
+	 */
+	requiresFullBuild() {
+		if (this.builder.forceCleanBuild) {
+			this.logger.trace('Full build required, force clean build flag is set.');
+			return true;
+		}
+
+		const currentAnalyzeOptionsHash = this.generateHash(JSON.stringify(this.defaultAnalyzeOptions));
+		if (!this.data.analyzeOptionsHash || this.data.analyzeOptionsHash !== currentAnalyzeOptionsHash) {
+			this.logger.trace('Full build required, jsanalyze options changed.');
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Generates SHA-1 hash from the given string value.
+	 *
+	 * @param {String} value String value to hash.
+	 * @return {String}
+	 */
+	generateHash(value) {
+		return crypto.createHash('sha1').update(value).digest('hex');
+	}
+
+	/**
+	 * Loads data from the previous task run.
+	 *
+	 * @return {Boolean} True if the data was sucessfully loaded, false if not.
+	 */
+	loadTaskData() {
+		if (!fs.exists(this.dataFilePath)) {
```

Comment:
does this even work properly? You're calling the async API but checking the result which is presumably a Promise (and therefore will always evaluate to "true")